packages/blade/src/components/DatePicker/BaseDatePicker.web.tsx

```diff
@@ -190,8 +195,16 @@ const BaseDatePicker = <Type extends DateSelectionType = 'single'>({
     fireNativeEvent(referenceRef, ['change']);
     handleReset();
     close();
+    setSelectedFilters((prev: string[]) => {
+      return prev.filter((filter) => filter !== label);
+    });
   };
 
+  useEffect(() => {
+    handleClear();
```

Comment:
Use effect will also run on mount and if user is controlling the component or need to initialize with some value this will clear the values on first mount.